word_search.py

```diff
@@ -0,0 +1,102 @@
+'''
+Created on 9 June 2017
+
+@author: Eva
+'''
+from queue import Queue
+
+class WordSearch:
+    
+    def __init__(self):
+        return
+        
+    def words_found_in_grid(self, rowNo, columnNo, grid, dictionary):
+        """Uses bfs to find valid words from a grid according to a given dictionary 
+        
+            Args:
+                rowNo: number of rows in the grid
+                columnNo: number of columns in the grid
+                grid: two-dimensional array of characters
```

Comment:
Yes, that's what the assignment requested. However, looking at the code, it seems that it is actually unnecessary to do so.

The reason this was phrased this way is probably that the person writing the assignment wants to make this apply to all languages (C++, Java, Python) and in C++ this is actually a requirement (and it is also a requirement depending on how you choose to implement your solution).

That said, it is not uncommon in practice to start with an idea of what one wants to build, but then realize that it should be done differently. For that reason, I felt it was worth mentioning this as part of the code review. In the end, the code is better without those parameters, so I think we should do so.

Does it make sense?